Bio/PDB/PDBList.py

```
@@ -701,88 +702,166 @@ def get_seqres_file(self, savefile="pdb_seqres.txt"):
+    )
+
+    # Add mutually exclusive arguments for mode
+    mode = parser.add_mutually_exclusive_group(required=True)
+    mode.add_argument(
+        "-update",
+        action="store_true",
+        help="write weekly PDB updates to local pdb tree",
+    )
+    mode.add_argument(
+        "-all", action="store_true", help="write all PDB entries to local pdb tree"
+    )
+    mode.add_argument(
+        "-obsol",
+        action="store_true",
+        help="write all obsolete PDB entries to local pdb tree",
+    )
+    mode.add_argument(
+        "-assemb",
+        action="store_true",
+        help="write all assemblies for each PDB entry to local pdb tree",
+    )
+    mode.add_argument(
+        "-single", help="retrieve single structure - accepts a single PDB code"
+    )
+    mode.add_argument(
+        "-multi",
+        help="retrieve a set of structures - accepts a comma separated list of PDB codes",
+    )
+
+    # Add mututally exclusive arguments for format
+    format = parser.add_mutually_exclusive_group()
+    format.add_argument(
+        "-pdb", action="store_true", help="downloads structures in PDB format"
+    )
+    format.add_argument(
+        "-xml", action="store_true", help="downloads structures in PDBML (XML) format"
+    )
+    format.add_argument(
+        "-mmtf", action="store_true", help="downloads structures in mmtf format"
+    )
+
+    # Add final options
+    others = parser.add_argument_group()
+    others.add_argument(
+        "-d",
+        action="store_true",
+        help="a single directory will be used as pdb_path, not a tree",
+    )
+    others.add_argument(
+        "-o", action="store_true", help="overwrite existing structure files"
+    )
+    others.add_argument(
+        "-with-assemblies",
+        action="store_true",
+        help="downloads assemblies along with regular entries",
+    )
+
+    # Parse arguments
+    args = parser.parse_args()
```

I like to have the help when I do not use a CLI correctly 
```suggestion
    try:
        args = parser.parse_args()
    except:
        args.print_help()
        sys.exit(0)
```
question of style one more time I guess 😄

---

I think including something like this is great, but as it is currently written it doesn't work because if `parser.parse_args()" fails there is no `args` object to tap into. I would also want to make sure we avoid the bare except if we can.

---

yes, I mean `parser.print_help()`, my bad

---

Still not entirely sure if this is necessary. The default behaviour when the `.parse_args()` fails is to print the usage message along with what error occurred. I think if the user wants to see the full help output they can call it themselves, it just seems a bit verbose to spit it all out in one go.

If there is consensus that this is behaviour that we want I will just need to stop `argparse` from printing the usage/error message because it's giving redundant information right now.

e.g.

```
❯ python PDBList.py -updaaa
usage: PDBList.py [-h] (-update | -all | -obsol | -assemb | -codes ) [-format {pdb,xml,mmtf}] [-d] [-o] [-with-assemblies] [pdb_path]
PDBList.py: error: one of the arguments -update -all -obsol -assemb -codes is required
usage: PDBList.py [-h] (-update | -all | -obsol | -assemb | -codes ) [-format {pdb,xml,mmtf}] [-d] [-o] [-with-assemblies] [pdb_path]

    (c) Kristian Rother 2003, Wiktoria Karwicka & Jacek Smietanski 2016
    Contributed to Biopython
	...
```